akka23-conductr-client-lib/src/main/scala/com/typesafe/conductr/clientlib/akka/BundlesConnector.scala

```diff
@@ -81,56 +165,111 @@ object BundlesConnector {
    * having received events because then startup could fail if there are none i.e. we must always
    * get the initial state of bundle info.
    */
-  private def consumeBundlesEvents(connector: ActorRef, getBundlesEvents: BundlesConnector.GetBundleEvents)(
-    implicit
-    ec: ExecutionContext, mat: ActorMaterializer
-  ): Unit =
-    for {
-      eventStreamResult <- getBundlesEvents()
-    } eventStreamResult match {
-      case EventStreamSuccess(sseSource) =>
-        sseSource
-          .runForeach(connector ! _)
-          .map(done => connector ! ConsumeBundlesEventsOutcome(Success(done)))
-          .recover {
-            case error =>
-              connector ! ConsumeBundlesEventsOutcome(Failure(error))
+  private def consumeBundlesEvents(connector: ActorRef, connection: Connection, apiVersion: ApiVersion.Value)(implicit ec: ExecutionContext, mat: ActorMaterializer): Unit = {
+    import EventStreamUnmarshalling._
+
+    Source
+      .single(bundlesEventsRequest(apiVersion))
+      .via(connection)
+      .alsoTo(
+        Sink.head.mapMaterializedValue(_.foreach { response =>
+          if (response.status.isSuccess())
+            connector ! GetBundles
+        })
+      )
+      .mapAsync(1) { response =>
+        Unmarshal(response.entity.withSizeLimit(Long.MaxValue))
```

Comment:
Actually, now I remember - there's a bug within Akka SSE where the size limit validation doesn't get disabled when reading the incoming SSE response:

https://github.com/hseeberger/akka-sse/issues/101

This will cause the SSE connection to be terminated when the default 8MB size limit threshold is reached.

The BundlesConnector will be able to cope with this Actually, now I remember - there's a bug within Akka SSE where the size limit validation doesn't get disabled when reading the incoming SSE response:

https://github.com/hseeberger/akka-sse/issues/101

This will cause the SSE connection to be terminated when the default 8MB size limit threshold is reached.

The BundlesConnector is still able to cope with this since it will reconnect to the bundle events endpoint, but during the re-connection it's possible to miss some events.